sdk/azidentity/managed_identity_client.go

```diff
@@ -254,13 +248,21 @@ func (c *managedIdentityClient) authenticate(ctx context.Context, id ManagedIDKi
 	}
 
 	if azruntime.HasStatusCode(resp, http.StatusOK, http.StatusCreated) {
-		return c.createAccessToken(resp)
+		tk, err := c.createAccessToken(resp)
+		if err != nil && c.chained && c.msiType == msiTypeIMDS {
```

Comment:
It's not clear to me why we only want to do this if the credential is part of a chain. If we return the unmarshaling error, is that not surfaced to the caller?

Reply:
The caller gets an error in any case. The type matters though. createAccessToken returns AuthenticationFailedError, and ChainedTokenCredential will stop iterating when it gets one of those. I want the chain to proceed to its next credential, so I return credentialUnavailableError instead when this credential's part of a chain. I'm afraid to do that unconditionally because someone may have code like this:
```go
cred, _ := azidentity.NewManagedIdentityCredential(nil)
client := azfoo.New(cred)
err := client.Frobulate()
if err != nil {
	var e *azidentity.AuthenticationFailedError
	if errors.As(err, &e) {
		// ...
	}
}
```

Reply:
So if `authenticate()` returns an arbitrary error, and that error isn't a `credentialUnavailableError`, it's surfaced as an `AuthenticationFailedError`, is that right?

Reply:
Yes, that happens elsewhere:
https://github.com/Azure/azure-sdk-for-go/blob/bf1e31b5690cbd12f8bfdc5a7926782c4c7e84eb/sdk/azidentity/confidential_client.go#L114-L115